pandas/tests/frame/test_ufunc.py

```diff
@@ -60,6 +62,25 @@ def test_binary_input_dispatch_binop(dtype):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "func,expected",
+    [
+        (np.add, [2, 3, 4, 5]),
+        (partial(np.add, where=[[False, True], [True, False]]), [0, 3, 4, 0]),
+        (np.power, [1, 2, 3, 4]),
+    ],
+)
+def test_ufunc_passes_args(frame_or_series, func, expected):
+    # GH#40662
+    arr = np.array([[1, 2], [3, 4]])
+    df = pd.DataFrame(arr)
+    result = np.zeros_like(df)
+    func(df, 1, out=result)
+
+    expected = np.array(expected).reshape(2, 2)
+    tm.assert_numpy_array_equal(result, expected)
```

Comment:
can you also test the return type/values of the operation itself.

Reply:
done